galaxy_skeleton1.R

```diff
@@ -2,24 +2,138 @@
+  data_raw_1[(L+1):length(data_raw_1)]%>% 
+  gsub("\\|", ",", .) %>% 
+  gsub(" ", "", .)
+
+comma_separated_values_with_names <- 
+  c(paste(variable_names, collapse = ","),
+    comma_separated_values)
+
+#write into csv file
+cat(comma_separated_values_with_names, sep = "\n", file = "data_1.csv")
+
+# Read the finished .csv back into R in the normal way.
+galaxies_1 <- read_csv("data_1.csv")
+
+#Problem 3
+#-------------------------------------------------------------------------------
+# Histogram
+library(ggplot2)
+
+ggplot(galaxies_1, aes(x = a_26)) +
+  geom_histogram(bins = 25, fill ="gray", color="black") +
+  labs(title = "Galaxy Size Distribution",
+       x = "Linear diameter of the galaxy in kpc (a_26)",
+       y = "Frequency") +
+  theme_minimal()
+
+# Observe lower frequency of lower size than larger size
+# Assuming an more equal distribution between smaller and larger sizes, there 
+# should not be a large peak
+
+# A plausible explanation could be that the sample is affected by selection bias.
+# Smaller ones could be more difficult to detect the further away from us they 
+# are which would exclude them from the sample of detected ones.
+
+#Problem 4
+#-------------------------------------------------------------------------------
+#P1 - Redo operations from P2 with new data
+data_raw_2 <- readLines(con="UCNG_Table4.txt")
+
+substr(x = data_raw_2, start = 2, stop = 3)
+
+L <- 
+  (substr(x = data_raw_2, start = 2, stop = 3) == "--") %>% 
+  which(arr.ind = TRUE) %>% 
+  min()
+
+variable_names <- 
+  str_split(string = data_raw_2[(L-1)] , pattern = "\\|") %>% 
+  unlist() %>% 
+  str_trim()
+
+comma_separated_values <- 
+  data_raw_2[(L+1):length(data_raw_2)]%>% 
+  gsub("\\|", ",", .) %>% 
+  gsub(" ", "", .)
+
+comma_separated_values_with_names <- 
+  c(paste(variable_names, collapse = ","),
+    comma_separated_values)
+
+#write into csv file
+cat(comma_separated_values_with_names, sep = "\n", file = "data_2.csv")
+
 # Read the finished .csv back into R in the normal way.
+galaxies_2 <- read_csv("data_2.csv")
+
+# Select and merge relevant data from the two datasets
+galaxies_1 <- galaxies_1 |> select(name, a_26, D, md)
+galaxies_2 <- galaxies_2 |> select(name, cz)
+
+galaxies_1 <- galaxies_1 |> full_join(galaxies_2)
+
+# Create plot
+ggplot(galaxies_1) +
+  geom_point(aes(x = D, y = cz)) +  # Create a scatterplot
+  labs(title = "Galaxies: Distance vs. Velocity",
+       x = "Distance",
+       y = "Velocity") +
+  geom_smooth(aes(x = D, y = cz),method = "lm", se = FALSE, color = "red",linetype=2) +
+  theme_minimal()
+
+#We observe a positive correlation between distance and velocity indicating
+#what Hubble in accordance with what Hubble found
+
+#P2
+#Convert km/h to km/s
+galaxies_1 <- galaxies_1 |> mutate(cz1 = cz/3600)
 
+#Calculate hubbles constant estimate
+hubbles_constant <- 
+  mean(galaxies_1$cz1, na.rm = T) / mean(galaxies_1$D, na.rm = T)
+#Hubbles constant is estimated to be 0.02km/s/Mpc
 
+hubbles_constant
```

Comment:
Hubbles constant can be estimated using a linear regression with Velocity as dependent variable, distance as independent variable and removing the intercept.

The constant is the estimated to be the coefficient of Distance.